pkg/registry/generic/rest/proxy.go

```diff
@@ -174,6 +174,12 @@ func (h *UpgradeAwareProxyHandler) tryUpgrade(w http.ResponseWriter, req *http.R
 func (h *UpgradeAwareProxyHandler) dialURL() (net.Conn, error) {
 	dialAddr := netutil.CanonicalAddr(h.Location)
 
+	if h.Transport != nil {
```

Comment:
The check for nil is redundant with the check for casting. This can be compressed into

```
if httpTransport, ok := h.Transport.(*http.Transport); ok && httpTransport.Dial != nil {
  return ...
}
```